azure-iot-device/tests/iothub/shared_client_tests.py

```diff
@@ -410,6 +445,34 @@ def test_sastoken(self, mocker, client_class):
             hostname=self.hostname, device_id=self.device_id
         )
 
+        custom_ttl = 1000
+        client_class.create_from_symmetric_key(
+            symmetric_key=self.symmetric_key,
+            hostname=self.hostname,
+            device_id=self.device_id,
+            sastoken_ttl=custom_ttl,
+        )
+
+        # SymmetricKeySigningMechanism created using the provided symmetric key
+        assert sksm_mock.call_count == 1
+        assert sksm_mock.call_args == mocker.call(key=self.symmetric_key)
+
+        # SasToken created with the SymmetricKeySigningMechanism, the expected URI, and the custom ttl
+        assert sastoken_mock.call_count == 1
+        assert sastoken_mock.call_args == mocker.call(
+            expected_uri, sksm_mock.return_value, ttl=custom_ttl
+        )
+
+    @pytest.mark.it(
+        "Uses 3600 seconds (1 hour) as the default SasToken TTL if no custom TTL is provided"
+    )
+    def test_sastoken_default(self, mocker, client_class):
+        sksm_mock = mocker.patch.object(auth, "SymmetricKeySigningMechanism")
+        sastoken_mock = mocker.patch.object(st, "SasToken")
+        expected_uri = "{hostname}/devices/{device_id}".format(
+            hostname=self.hostname, device_id=self.device_id
+        )
```

Comment:
We test SasToken config here, as opposed in the shared user option tests, because this isn't a kwarg that goes to the PipelineConfig. I'll probably rename `SharedIoTHubClientCreateMethodUserOptionTests` down the road to reflect that